packages/gafl-webapp-service/src/handlers/__tests__/analytics-handler.spec.js

```diff
@@ -18,19 +18,38 @@ describe('The analytics handler', () => {
   })
 
   it.each([
-    ['https://localhost:3000', 'https://localhost:3000/buy/name', '/buy/name'],
-    ['https://localhost:1234', 'https://localhost:1234/example/test/redirect', '/example/test/redirect'],
-    ['https://testserver-example-fish', 'https://testserver-example-fish/buy/renew/identify', '/buy/renew/identify']
-  ])('redirects to correct page', async (origin, referer, redirect) => {
+    ['https://localhost:3000', 'https://localhost:3000/buy/name', 'https://localhost:3000', '/buy/name'],
+    ['https://localhost:1234', 'https://localhost:1234/example/test/redirect', 'https://localhost:1234', '/example/test/redirect'],
+    [
+      'https://testserver-example-fish',
+      'https://testserver-example-fish/buy/renew/identify',
+      'https://testserver-example-fish',
+      '/buy/renew/identify'
+    ]
+  ])('redirects to correct page if url matches header', async (origin, referer, host, redirect) => {
     const payload = {
       analyticsResponse: 'accept'
     }
-    const request = generateRequestMock(payload, 'analytics', origin, referer)
+    const request = generateRequestMock(payload, 'analytics', origin, referer, host)
     const responseToolkit = generateResponseToolkitMock()
     await analyticsHandler(request, responseToolkit)
     expect(responseToolkit.redirect).toHaveBeenCalledWith(addLanguageCodeToUri(request, redirect))
   })
 
+  it.each([
+    ['https://localhost:3000', 'https://localhost:3000/buy/name', 'https://localhost:3047'],
+    ['https://localhost:1234', 'https://localhost:1234/example/test/redirect', 'https://notsamehost:1234'],
+    ['https://testserver-example-fish', 'https://testserver-example-fish/buy/renew/identify', 'https://hdfhdskfhs-ghj-vgjh']
+  ])('redirects to correct page if url matches header', async (origin, referer, host) => {
+    const payload = {
+      analyticsResponse: 'accept'
+    }
+    const request = generateRequestMock(payload, 'analytics', origin, referer, host)
+    const responseToolkit = generateResponseToolkitMock()
+    await analyticsHandler(request, responseToolkit)
+    expect(responseToolkit.redirect).toHaveBeenCalledWith(addLanguageCodeToUri(request, '/buy'))
+  })
```

Comment:
This test and the previous one both have the same name, 'redirects to correct page if url matches header'. I think you need to describe them both better. In this second set of tests it looks like you're testing the default, so state that in the test name - something like 'redirects to /buy if the HTTP_REFERER host and port does not match the host and port of the current page'

Furthermore, I think the `addLanguageCodeToUri` is muddying the water. You should mock `addLanguageCodeToUri` so it returns whatever is sent to it without modifying it, then your assertion could omit `addLanguageCodeToUri` entirely and do `toHaveBeenCalledWith('/buy')`. Do this for the first test case too. We're not testing addLanguageCodeToUri, so it only causes confusion.